File: app/src/main/java/com/mohnage7/movies/base/DataWrapper.java
package com.mohnage7.movies.base;

public class DataWrapper<T> {
    private T data;
    private BaseError baseError;

    public DataWrapper(T data, BaseError baseError) {
        this.data = data;
        this.baseError = baseError;
    }

    public T getData() {
        return data;
    }

    public BaseError getBaseError() {
        return baseError;
    }
}

File: app/src/main/java/com/mohnage7/movies/network/model/DataWrapper.kt
package com.mohnage7.movies.network.model

class DataWrapper<T>(val status: Status, val data: T?, val message: String?) {

    enum class Status {
        SUCCESS, ERROR, LOADING
    }

    companion object {

        fun <T> success(data: T): DataWrapper<T> {
            return DataWrapper(Status.SUCCESS, data, null)
        }

        fun <T> error(msg: String, data: T?): DataWrapper<T> {
            return DataWrapper(Status.ERROR, data, msg)
        }

        fun <T> loading(data: T?): DataWrapper<T> {
            return DataWrapper(Status.LOADING, data, null)
        }
    }

}
